src/compiler/moduleNameResolver.ts

```diff
@@ -1097,13 +1097,18 @@ namespace ts {
     export function getPackageNameFromAtTypesDirectory(mangledName: string): string {
         const withoutAtTypePrefix = removePrefix(mangledName, "@types/");
         if (withoutAtTypePrefix !== mangledName) {
-            return stringContains(withoutAtTypePrefix, mangledScopedPackageSeparator) ?
-                "@" + withoutAtTypePrefix.replace(mangledScopedPackageSeparator, ts.directorySeparator) :
-                withoutAtTypePrefix;
+            return getPackageNameFromAtTypesDirectoryWithoutPrefix(withoutAtTypePrefix);
         }
         return mangledName;
     }
 
+    /* @internal */
+    export function getPackageNameFromAtTypesDirectoryWithoutPrefix(withoutAtTypePrefix: string): string {

```

Comment:
I would call this `unmangleTypesPackageName`, and the parameter `typesPackageName`.

Reply:
I went with `getUnmangledNameForScopedPackage` for consistency with `getMangledNameForScopedPackage`.